sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -427,6 +427,11 @@ private[hive] class HiveClientImpl(
   }
 
   override def alterTable(tableName: String, table: CatalogTable): Unit = withHiveState {
+    // Not allowed to switch the table type
+    if (table.properties.contains("EXTERNAL")) {
```

Comment:
this check doesn't work for `InMemoryCatalog`, we should think about a center place for this kind of checks